src/plugins/data/common/search/aggs/types.ts

```diff
@@ -80,6 +81,17 @@ export interface AggsCommonSetup {
 /** @internal */
 export interface AggsCommonStart {
   calculateAutoTimeExpression: ReturnType<typeof getCalculateAutoTimeExpression>;
+  /**
+   * Helper function returning meta data about use date intervals for a data table column.
+   * If the column is not a column created by a date histogram aggregation of the esaggs data source,
+   * this function will return undefined.
+   *
+   * Otherwise, it will return the following attributes in an object:
+   * * `timeZone` time zone used to create the buckets (important e.g. for DST),
+   * * `timeRange` total time range of the fetch data (to infer partial buckets at the beginning and end of the data)
+   * * `interval` Interval used on elasticsearch (`auto` resolved to the actual interval)
+   */
+  getDateMetaByDatatableColumn: ReturnType<typeof getDateMetaByDatatableColumn>;
```

Comment:
do we need this on the contract or can it be stateless ?

Reply:
IMHO it makes sense on the contract as it depends on so many small stateful bits: https://github.com/elastic/kibana/pull/81007/files/6b397a270a3acadec928245e7193542784eab18f#diff-18c386e256307710b223df056a4456567098685d3570e69455d66b258bdcfb2aR94

If the consumer has to collect of all that, it gets much harder to use (especially as a lot of it is already quite close in code)

Reply:
seems to me its two settings we pass in + index pattern which i still want to figure out what it does and how we can clean that up. but after thinking thru this some more i agree to leave this on the contract as even if this function might not require much state others will definetely do (to get field formatter instance or agg config instance)